File: frontend/src/app/models/PersonalBook.ts
import {Book} from "./Book";

export class PersonalBook {
  constructor(
    public isbn: number,
    public status: string,
    public estimation?: number,
    public comment?: string
  ) {
  }
}

export interface PersonalBookImpl extends PersonalBook {
  bookDtoImpl: Book
}

export enum Status {
  WILL_READ = 'Буду читать',
  READING = 'Читаю',
  HAVE_READ = 'Прочитано'
}

File: backend/server/src/main/java/server/entities/PersonalBook.java
package server.entities;

import lombok.Getter;
import lombok.Setter;

import jakarta.persistence.*;
import java.io.Serializable;

@Entity
@Getter
@Setter
@Table(name = "personal_books", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"isbn", "phone", "email"})
})
public class PersonalBook extends DefaultEntity implements Serializable {

    private String isbn;
    private String phone;
    private String email;
    private String status;
    private Integer estimation;
    private String comment;
    @ManyToOne
    @JoinColumn(name = "isbn", referencedColumnName = "isbn", insertable = false, updatable = false)
    private Book book;

    @ManyToOne
    @Transient
    @JoinColumns(
            {
                    @JoinColumn(name = "phone", referencedColumnName = "phone", insertable = false, updatable = false),
                    @JoinColumn(name = "email", referencedColumnName = "email", insertable = false, updatable = false)
            }
    )
    private User user;

}
